core/coreapi/coreapi.go

```diff
@@ -244,7 +244,11 @@ func (api *CoreAPI) WithOptions(opts ...options.ApiOption) (coreiface.CoreAPI, e
 
 	if settings.Offline || !settings.FetchBlocks {
 		subAPI.exchange = offlinexch.Exchange(subAPI.blockstore)
-		subAPI.blocks = bserv.New(subAPI.blockstore, subAPI.exchange)
+		var bsopts []bserv.Option
+		if cfg.Datastore.WriteThrough {
+			bsopts = append(bsopts, bserv.WriteThrough())
+		}
+		subAPI.blocks = bserv.New(subAPI.blockstore, subAPI.exchange, bsopts...)
```

Comment:
This is ok, but is might be nicer to have a blockservice option `WithWriteThrough` that takes a `bool`. Then to code becomes:
```go
subAPI.blocks = bserv.New(subAPI.blockstore, subAPI.exchange,
    bserv.WithWriteThrough(cfg.Datastore.WriteThrough))
```

Reply:
Right? We need to change that in boxo. I can take care, but we can merge this and fix that later for next boxo release.